src/database.js

```diff
@@ -0,0 +1,88 @@
+import fs from 'fs';
+import sqlite3 from 'sqlite3';
+
+const schema = fs.readFileSync('./src/schema.sql').toString();
+const connect = new Promise((resolve, reject) => {
+  const result = new sqlite3.Database('./.data/application.db', (error) => {
+    if (error) {
+      reject(error);
+    } else {
+      console.log('i have hereby connected :)')
+      result.exec(schema, (execError) => {
+        if (execError) {
+          reject(execError);
+        } else {
+          resolve(result);
+        }
+      });
+    }
+  });
+});
+
+const query = (method) => async (...args) => {
+  const db = await connect;
+  return new Promise((resolve, reject) => {
+    db[method](...args, (error, result) => {
+      if (error) {
+        reject(error);
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
+
+const run = query('run');
+const get = query('get');
+const all = query('all');
+
+export const getSetting = async (name) => {
+  const result = await get('select value from settings where name = ?', name);
+
+  if (!result) {
+    return null;
+  }
+
+  const { value } = result;
+
+  if (typeof value === 'string') {
+    return JSON.parse(value);
```

Comment:
Every value in the settings table is assumed to be serialized as JSON. I'm opting not to actually use SQLite's built-in JSON functions for this, though, because we'll never want to treat the values as anything but blobs of text at the database level. The only reason I can think of to actually use native JSON is if we can get multiple rows deserialized all at once somehow—I'll check if this is an option.